### PR TITLE
timerdevice_clock_drift_with_sleep: fix bugs

### PR DESCRIPTION
**Pull request by dinghuiqing**
1. Remove "no up" in cfg file and check host cpu num to get vcpu num
    2. Fix the expected result, the elapsed time should > 10s and < 11s
    3. Add definition for "host_cpu_cnt_cmd" in cfg file

ID: 1468145

Signed-off-by: Huiqing Ding <huding@redhat.com>

### CODE REVIEW

File: qemu/tests/timerdevice_clock_drift_with_sleep.py [hunk 1]
```diff
@@ -38,36 +38,37 @@ def verify_elapsed_time():
...
         if times_list:
-            raise error.TestFail("Unexpected time drift found:"
-                                 " Detail: '%s'" % output)
+            test.fail("Unexpected time drift found:"
+                      " Detail: '%s' \n timelist: %s" % (output, times_list))
 
-    error.context("Sync the host system time with ntp server", logging.info)
-    utils.system("yum install -y ntpdate; ntpdate clock.redhat.com")
+    error_context.context("Sync the host system time with ntp server", logging.info)
+    ntp_sync_cmd = "yum install -y ntpdate; ntpdate clock.redhat.com"
```
Comment (balamuruhans) [inline qemu/tests/timerdevice_clock_drift_with_sleep.py]: can we use `utils_package` to handle package installation ? it is handled in better way.

File: qemu/tests/timerdevice_clock_drift_with_sleep.py [hunk 2]
```diff
@@ -98,36 +99,37 @@ def verify_elapsed_time():
...
+    error_context.context("Pin every vcpu to physical cpu", logging.info)
     host_cpu_cnt_cmd = params["host_cpu_cnt_cmd"]
-    host_cpu_num = utils.system_output(host_cpu_cnt_cmd).strip()
+    host_cpu_num = process.system_output(host_cpu_cnt_cmd, shell=True).strip()
     host_cpu_list = (_ for _ in range(int(host_cpu_num)))
     cpu_pin_list = zip(vm.vcpu_threads, host_cpu_list)
     if len(cpu_pin_list) < len(vm.vcpu_threads):
-        raise error.TestNAError("There isn't enough physical cpu to"
-                                " pin all the vcpus")
+        test.error("There isn't enough physical cpu to"
```
Comment (balamuruhans) [inline qemu/tests/timerdevice_clock_drift_with_sleep.py]: should this be `test.cancel()`

File: qemu/tests/cfg/timerdevice.cfg [hunk 3]
```diff
@@ -9,15 +10,13 @@
             msr_tools_cmd = "a=$(rdmsr -d 0x00000010); echo -e ibase=16\\n $a  | tail -n 1 | while read n; do wrmsr 0x00000010 $(($n+100000000000)); echo $n; done"
         - clock_drift_with_sleep:
             only RHEL
-            no up
             type = timerdevice_clock_drift_with_sleep
             rtc_base = utc
             rtc_clock = host
             rtc_drift = slew
```
Comment (vivianQizhu) [inline qemu/tests/cfg/timerdevice.cfg]: Looks this does not work for powerpc?

File: qemu/tests/timerdevice_clock_drift_with_sleep.py (hunk 2 above)
Comment (vivianQizhu) [reply qemu/tests/timerdevice_clock_drift_with_sleep.py]: Agree, and if so maybe better move the judgement to the beginning of the test case .

File: qemu/tests/cfg/timerdevice.cfg [hunk 4]
```diff
@@ -9,15 +10,13 @@
             msr_tools_cmd = "a=$(rdmsr -d 0x00000010); echo -e ibase=16\\n $a  | tail -n 1 | while read n; do wrmsr 0x00000010 $(($n+100000000000)); echo $n; done"
         - clock_drift_with_sleep:
             only RHEL
-            no up
```
Comment (sitoliu) [inline qemu/tests/cfg/timerdevice.cfg]: In this case, vcpu needs to be pinned to specific host cpus, thus we need a `smp` host, I guess `no up` makes sense.

**Pull request closed without merging**